File: src/systems/origin/apps/service/src/lib/normalizePath.test.ts
import { describe, expect, it } from 'vitest';
import { normalizePath } from './normalizePath';

describe('normalizePath', () => {
  it('resolves parent and current directory segments', () => {
    expect(normalizePath('foo/../bar/./baz')).toBe('/bar/baz');
    expect(normalizePath('/a/b/../c')).toBe('/a/c');
  });

  it('returns root for empty paths', () => {
    expect(normalizePath('')).toBe('/');
    expect(normalizePath('/')).toBe('/');
  });
});

File: src/systems/origin/apps/code-bucket/pkg/util/normalizePath_test.go
package util

import "testing"

func TestNormalizePath(t *testing.T) {
	if got := NormalizePath("foo/../bar/./baz"); got != "/bar/baz" {
		t.Fatalf("NormalizePath() = %q, want /bar/baz", got)
	}

	if got := NormalizePath(""); got != "/" {
		t.Fatalf("NormalizePath(\"\") = %q, want /", got)
	}

	if got := NormalizePath("/a/b/../c"); got != "/a/c" {
		t.Fatalf("NormalizePath() = %q, want /a/c", got)
	}
}
